compiler/src/dotty/tools/dotc/reporting/trace.scala

```diff
@@ -7,34 +7,35 @@ import config.Config
 import config.Printers
 import core.Mode
 
-object trace {
+abstract class TraceSyntax {
+  val isForced: Boolean
```

Comment:
Use a field `TraceSyntax(isForced: Boolean)`? I would expect it is cheaper

Reply:
Could you be more clear on what it is you call a field? The goal here is to override `isForced` with a `final val`, which should be guaranteed to be inlined whenever referenced, which would then mean that we get rid of useless branches in the inlined logging methods. This is somewhat moot, since `forceInline` does not seem to inline nested methods properly ( see https://github.com/lampepfl/dotty/pull/5738#issuecomment-469240942 ), but I don't see how making `isForced` a constructor argument makes it any cheaper.

Reply:
Here the `final val` implements an abstract one. It will be compiled to a virtual method call, I don't think the compiler can inline in this case. If `isForced` is a constructor argument, it will be compiled to a field. A field access is much cheaper than a virtual method call.

Reply:
I see the reasoning, but all the methods that access `isForced` are `@forceInline`, so after the body is inlined the compiler should see that the member being accessed is a `final val` and therefore is inlineable, no? Anyway, this PR has no impact on performance _and_ after we bootstrap we will probably rewrite all the methods in this class to `inline`, which will guarantee that they are inlined the way we want.

Side note: I am bookmarking this discussion to show the next person that tries to convince me it's a good idea to rely on the optimizer for inlining.